functions/go/set-labels/transformer/labels_test.go

```diff
@@ -0,0 +1,332 @@
+package transformer
```

Comment:
Some general suggestions for writing go unittests. 
- using for loop to parameterize the test cases.   See below example
- use some testdata generator method to reuse/simplify the code. See example `generateResourceList` below
example 1
```go
testdata := map[string]struct{
     resourcelist string
     expected string
} { 
      ""add labels to selector": {
           resourcelist: generateResourceList(cm, services)
          expected: ...
      } 
       ....
}

func generateResourceList(functionConfig string, singleItem string) {
        data := fmt.Sprintf("functionConfig: %s\nitems: \n... %s", functionConfig, singleItem)
        rl, _ := fn.ParseResourceList(data)
        return rl 
}
```

Reply:
Using for loop means there is only one unit test to run all the tests? I think it might not be good for debugging. I should reuse the code and I will fix that part.

Reply:
that's true, that's why the sample code uses the map key which I expect to describe the test case.
This is more or less a go convention for test parameterization.